RoboJackets/igvc-software
# Fix Navigation Client Issues

# Description

Fixes two issues with `navigation_client`

This PR does the following:
- Adds a new launch file to `igvc_navigation` called `set_waypoint_file_path.launch` which takes an arg `file_path` and sets the `waypoint_file_path` parameter to the value of `file_path`.
- Changes `simulation.launch` to use `set_waypoint_file_path.launch`.
- Fixes `NavigationClient::parseWaypointFile` not checking for comments or empty lines.

Fixes #621 and #622 

# Testing steps (If relevant)
## Test Case 1
1. run `roslaunch igvc_navigation set_waypoint_file_path.launch folder_path:=abc file_name:=def`
2. run `rosparam get /waypoint_file_path

Expectation: `abc/def` is displayed.

## Test Case 2
1. run `roslaunch igvc_navigation set_waypoint_file_path.launch file_name:=abc`
2. run `rosparam get /waypoint_file_path`

Expectation: `$PATH_TO_IGVC_SANDBOX/waypoints/abc` is displayed.

## Test Case 3
1. run `roslaunch igvc_navigation navigation_client.launch` while a simulation that uses a waypoint file with comments and clear lines.

Expectation: navigation_client does not break while loading waypoints

# Self Checklist
